DynamicModeler/Logic/vtkSlicerDynamicModelerExtrudeTool.cxx

```diff
@@ -104,32 +108,45 @@ vtkSlicerDynamicModelerExtrudeTool::vtkSlicerDynamicModelerExtrudeTool()
   /////////
   // Parameters
 
-  ParameterInfo parameterExtrusionScale(
-    "Extrusion scale",
-    "Length of normals, line or point distance is scaled by this value to get the offset surface position. Use negative value for reversing extrusion direction.",
-    "ExtrusionScale",
+  ParameterInfo parameterLengthMode(
+    "Length mode",
+    "Can be fixed length or scaled length. Scaled mode is ignored for planes",
+    EXTRUDE_LENGTH_MODE,
+    PARAMETER_STRING_ENUM,
+    "Fixed");
+
+  vtkNew<vtkStringArray> possibleValues;
+  parameterLengthMode.PossibleValues = possibleValues;
+  parameterLengthMode.PossibleValues->InsertNextValue("Fixed");
+  parameterLengthMode.PossibleValues->InsertNextValue("Scaled");
+  this->InputParameterInfo.push_back(parameterLengthMode);
+
+  ParameterInfo parameterExtrusionValue(
+    "Extrusion value",
+    "Value used for fixed length or multiplied by point position. Use negative value for reversing extrusion direction.",
+    EXTRUDE_VALUE,
     PARAMETER_DOUBLE,
     1.0);
-  this->InputParameterInfo.push_back(parameterExtrusionScale);
+  this->InputParameterInfo.push_back(parameterExtrusionValue);
 
   this->InputModelToWorldTransformFilter = vtkSmartPointer<vtkTransformPolyDataFilter>::New();
   this->InputModelNodeToWorldTransform = vtkSmartPointer<vtkGeneralTransform>::New();
   this->InputModelToWorldTransformFilter->SetTransform(this->InputModelNodeToWorldTransform);
 
+  this->NormalsFilter = vtkSmartPointer<vtkPolyDataNormals>::New();
+  this->NormalsFilter->SetInputConnection(this->InputModelToWorldTransformFilter->GetOutputPort());
```

Comment:
We may already have good normals in the polydata, which `vtkPolyDataNormals` may overwrite with incorrect vectors (the filter may fail for complex shapes). If the polydata already has normals then it would be better not to overwrite them. We may do such things elsewhere, but it still should be avoided (and fixed the other occurrences as well).